### PR TITLE
implement wifmanager and update examples

### PR DESCRIPTION
**Pull request by jerryneedell**
added support for wifimanager
updated examples

### CODE REVIEW

File: adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+"""
+`adafruit_espatcontrol_wifimanager`
+================================================================================
+
+WiFi Manager for making ESP32 SPI as WiFi much easier
```
Comment (makermelissa) [inline adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py]: We should change the name here

File: adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+* Author(s): Melissa LeBlanc-Williams, ladyada
```
Comment (makermelissa) [inline adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py]: Don't forget to add your name to this

File: adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+# pylint: disable=no-name-in-module
+
+import adafruit_espatcontrol.adafruit_espatcontrol_requests as requests
+
+class ESPAT_WiFiManager:
+    """
+    A class to help manage the Wifi connection
+    """
+    def __init__(self, esp, secrets, status_pixel, attempts=2):
```
Comment (makermelissa) [inline adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py]: Not sure if this needs to be fixed in ESP32SPI too, but it should be: status_pixel=None

File: adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py.save [hunk 4]
```diff
@@ -0,0 +1,216 @@
+# The MIT License (MIT)
```
Comment (makermelissa) [inline adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py.save]: This looks like a duplicate file and should probably be removed from the commit.

File: adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py (hunk 1 above)
Comment (jerryneedell) [reply adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py]: done

File: adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py (hunk 2 above)
Comment (jerryneedell) [reply adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py]: done

File: adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py (hunk 3 above)
Comment (jerryneedell) [reply adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py]: done

File: adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py.save (hunk 4 above)
Comment (jerryneedell) [reply adafruit_espatcontrol/adafruit_espatcontrol_wifimanager.py.save]: good catch

**Comment by jerryneedell** [comment]
tested with Feather_m4_express and feather huzzah32

**Comment by jerryneedell** [comment]
also tested most functions on particle argon  -- not sure if we want to add particle argon examples but I thought it best to get this PR in and released before adding argon specific examples.

**Comment by makermelissa** [comment]
Is this supposed to work with the ESP8266? I haven't tried it, but I'm trying to figure out which combination to try wiring up.

**Comment by jerryneedell** [comment]
It should also work with ESP8266, but there will likely be more communication errors and retries.
I will do some testing

**Comment by makermelissa** [comment]
Is there firmware for the ESP32? Also, what is the wiring for say a Metro M4 and ESP32? I see the WiFi Co-Proc learning guide, but it has other hardware and I'm not quite sure what the equivalent to the GPIO0 is.

**Comment by jerryneedell** [comment]
posted AT firmwared file to discord

```
flash with 
/usr/local/bin/esptool.py --chip esp32 --port /dev/ttyUSB0 --baud 115200 --before default_reset --after hard_reset write_flash -z --flash_mode dio --flash_freq 40m --flash_size detect 0 factory_WROOM-32.bin 
pins - 
VUSB - VUSB
GND - GND
RX - TX
TX -RX
D5 - ESP32 RESET
D6 - ESP32 GPIO15

GPIO is not needed 
```

**Comment by makermelissa** [comment]
@jerryneedell, what is your confidence level on this? Should somebody else review it too or do you feel good about it?

**Comment by jerryneedell** [comment]
I'm confident that it it works OK and have no concerns with merging or releasing it. That said, I'm also happy to give more time for other reviews/ comments.  There is no rush.

**Comment by makermelissa** [comment]
Sounds good to me. I'd say let's merge it so people can start using it if they want.

**Pull request merged**